sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -182,7 +182,7 @@ class Analyzer(
 
```

Comment:
Could we remove `conf: SQLConf` in the constructor of `Analyzer`?

Reply:
> Could we remove `conf: SQLConf` in the constructor of `Analyzer`?

My original idea is to remove this parameter, but it's causing much more changing on the Test code. @cloud-fan , what's your thought?

Reply:
I think it's fine to keep it. The main goal is to let rules use `SQLConf.get`